### PR TITLE
`azurerm_netapp_snapshot`  waitLRO

### PR DESCRIPTION
**Pull request by shu-ying789**
test result:
snapp1

linked to:
https://github.com/hashicorp/terraform-provider-azurerm/issues/15641

### CODE REVIEW

File: internal/services/netapp/netapp_snapshot_resource.go [hunk 1]
```diff
@@ -187,9 +187,13 @@ func resourceNetAppSnapshotDelete(d *pluginsdk.ResourceData, meta interface{}) e
...
 	}
 
-	if _, err = client.Delete(ctx, id.ResourceGroup, id.NetAppAccountName, id.CapacityPoolName, id.VolumeName, id.Name); err != nil {
+	future, err := client.Delete(ctx, id.ResourceGroup, id.NetAppAccountName, id.CapacityPoolName, id.VolumeName, id.Name)
+	if err != nil {
 		return fmt.Errorf("deleting NetApp Snapshot %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
 	}
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of %q: %+v", id, err)
+	}
```
Comment (tombuildsstuff) [inline internal/services/netapp/netapp_snapshot_resource.go:196]: does this mean we can remove the WaitForState func below?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @shu-ying789 

Thanks for this PR - I've taken a look through and left a comment inline but if we can remove the WaitForState func then we should be able to take another look.

Thanks!

File: internal/services/netapp/netapp_snapshot_resource.go (hunk 1 above)
Comment (katbyte) [reply internal/services/netapp/netapp_snapshot_resource.go:196]: no, as per the service team maintainer it is still required

**Comment by katbyte** [comment]
closing as this will be fixed by #17043

**Pull request closed without merging**